#3531 [FIX] write website_published_need_sync is True on magento.product.template mapping if website_published is changed

wytzejan opened the pull request:
The moment website_published is changed on the product template or variant the newly introduced boolean website_published_need_sync in the Magento product template mapping becomes true.
This gives us the opportunity to only loop to the product templates with website_published_need_sync = True for product website published status updates to Magento.

wytzejan committed:
#3531 [FIX] write website_published_need_sync is True on magento.product.template mapping if website_published is changed

StefanRijnhart commented:
It's the webkul additions that could be improved. If you wrote them, consider to rewrite magento_bridge/mob_website_published_cron.py to new API, move the lookup of the store_ids and the related view_ids out of the loop.

wytzejan commented:
@StefanRijnhart Thanks for this input! I rewrote magento_bridge/mob_website_published_cron.py to the new API. The lookup of the store_ids and the related view_ids is completely removed out of the code. I have set the attribute 'status' as global setting in the Magento back-end so it is not longer needed to declare the status for each store view.
Please take a look at the updated code in our development environment. As mentioned I have changed the code of magento_bridge/mob_website_published_cron.py but I also changed the call to this method in magento_bridge/mob_synchronization.py (def sync_website_published)

StefanRijnhart commented:
Looking better now! It still contains the following, redundant line:
```
template_map_obj = template_map.browse(mapped_id.id)
```
Can you elimitate it?

wytzejan commented:
@StefanRijnhart You are right! I have elimitate this now. Please check the latest change in mob_website_published_cron.py on the development environment.

StefanRijnhart commented:
Great! Seems we both missed another `template_map.browse(mapped_id.id).write({` which I'll leave for now.
